ext/simdjson/src/bindings.cpp

```diff
@@ -0,0 +1,238 @@
+        case simdjson::dom::element_type::STRING :
+            ZVAL_STRING(&v, std::string_view(element).data());
+            break;
+        case simdjson::dom::element_type::INT64 : ZVAL_LONG(&v, int64_t(element));
+            break;
+        case simdjson::dom::element_type::UINT64 : ZVAL_LONG(&v, uint64_t(element));
+            break;
+        case simdjson::dom::element_type::DOUBLE : ZVAL_DOUBLE(&v, double(element));
+            break;
+        case simdjson::dom::element_type::BOOL :
+            ZVAL_BOOL(&v, bool(element));
+            break;
+        case simdjson::dom::element_type::NULL_VALUE :
+            ZVAL_NULL(&v);
+            break;
+        case simdjson::dom::element_type::ARRAY :
+            zval arr;
+            array_init(&arr);
+
+            for (simdjson::dom::element child : simdjson::dom::array(element)) {
+                zval value = create_array(child);
+                add_next_index_zval(&arr, &value);
+            }
+
+            v = arr;
+            break;
+        case simdjson::dom::element_type::OBJECT :
+            zval obj;
+            array_init(&obj);
+
+            for (simdjson::dom::key_value_pair field : simdjson::dom::object(element)) {
+                zval value = create_array(field.value);
+                add_assoc_zval_ex(&obj, field.key.data(), strlen(field.key.data()), &value);
+            }
+            v = obj;
+            break;
+        default:
+            break;
+    }
+
+    return v;
+}
+
+static zval create_object(simdjson::dom::element element)
+{
+    zval v;
+
+    switch (element.type()) {
+        //ASCII sort
+        case simdjson::dom::element_type::STRING :
+            ZVAL_STRING(&v, std::string_view(element).data());
+            break;
+        case simdjson::dom::element_type::INT64 : ZVAL_LONG(&v, int64_t(element));
+            break;
+        case simdjson::dom::element_type::UINT64 : ZVAL_LONG(&v, uint64_t(element));
+            break;
+        case simdjson::dom::element_type::DOUBLE : ZVAL_DOUBLE(&v, double(element));
+            break;
+        case simdjson::dom::element_type::BOOL :
+            ZVAL_BOOL(&v, bool(element));
+            break;
+        case simdjson::dom::element_type::NULL_VALUE :
+            ZVAL_NULL(&v);
+            break;
+        case simdjson::dom::element_type::ARRAY :
+            zval arr;
+            array_init(&arr);
+
+            for (simdjson::dom::element child : simdjson::dom::array(element)) {
+                zval value = create_object(child);
+                add_next_index_zval(&arr, &value);
+            }
+
+            v = arr;
+            break;
+        case simdjson::dom::element_type::OBJECT :
+            zval obj;
+            object_init(&obj);
+
+            for (simdjson::dom::key_value_pair field : simdjson::dom::object(element)) {
+                zval value = create_object(field.value);
+                add_property_zval_ex(&obj, field.key.data(), strlen(field.key.data()), &value);
```

Comment:
stdClass uses the standard write_property handlers. Using the ext/json/json_parser.y might be a bit more efficient 

```suggestion
            zend_object *o = Z_OBJ(obj);
            for (simdjson::dom::key_value_pair field : simdjson::dom::object(element)) {
                zval value = create_object(field.value);
                zend_std_write_property(o, zend_string_init(str, len, 0), &value, NULL);
```


Calling zend_hash_find and zend_hash_add_new directly on HASH_OF might be even faster since stdClass has no __set method, etc., but the additional performance gain is probably too small to be in scope.